src/source/Ice/SocketConnection.c

```diff
@@ -197,7 +199,25 @@ STATUS socketConnectionSendData(PSocketConnection pSocketConnection, PBYTE pBuf,
         }
 
         // sending through UDP never block
-        if (sendto(pSocketConnection->localSocket, pBuf, bufLen, 0, destAddr, addrLen) < 0) {
+        result = sendto(pSocketConnection->localSocket, pBuf, bufLen, 0, destAddr, addrLen);
+        if (result < 0 && (errno == EAGAIN || errno == EWOULDBLOCK)) {
+            FD_ZERO(&wfds);
+            FD_SET(pSocketConnection->localSocket, &wfds);
+            tv.tv_sec = 0;
+            tv.tv_usec = SOCKET_SEND_RETRY_TIMEOUT_MICRO_SECOND;
+            result = select(pSocketConnection->localSocket + 1, NULL, &wfds, NULL, &tv);
+
+            if (result > 0) {
+                result = sendto(pSocketConnection->localSocket, pBuf, bufLen, 0, destAddr, addrLen);
+            } else if (result == 0) {
+                DLOGD("select() timed out");
```

Comment:
Are these WARNS? I see you have DLOGD here but in the else below you have DLOGE. See whether this is indeed an DLOGE or DLOGW.

Also, in case of DLOGE you could use CHK_ERR.

Reply:
I think all these should be DLOGD cuz higher layer should to agnostic to send failures.

Reply:
That's what I thought. In that case, please make the other DLOGD